src/basic_recipes/contourf.jl

```diff
@@ -15,6 +15,8 @@ This can be used for example to draw bands for the upper 90% while excluding the
 In :normal mode, if you want to show a band from `-Inf` to the low edge, set `extendlow` to `:auto` for the same color as the first level, or specify a different color (default `nothing` means no extended band)
 If you want to show a band from the high edge to `Inf`, set `extendhigh` to `:auto` for the same color as the last level, or specify a different color (default `nothing` means no extended band)
 
+If you specify `levels` as an `AbstractVector{<:Real}`, note that the axis limits will be pulled back from the bounds of the data. This is to avoid unsightly zero margins between an irregular contour and the axis. To undo this behavior, you can use `tightlimits!(ax)` or the other axis limit functions.
```

Comment:
It's kind of the other way around, `tightlimits!` is only called when the limits are given as an integer https://github.com/JuliaPlots/Makie.jl/blob/master/src/makielayout/layoutables/axis.jl#L709-L713